lib/linked_list.rb

```diff
@@ -12,152 +12,276 @@ def initialize(value, next_node = nil)
 
 # Defines the singly linked list
 class LinkedList
-    def initialize
-      @head = nil # keep the head private. Not accessible outside this class
-    end
+  def initialize
+    @head = nil # keep the head private. Not accessible outside this class
+  end
 
-    # method to add a new node with the specific data value in the linked list
-    # insert the new node at the beginning of the linked list
-    # Time Complexity:  
-    # Space Complexity
-    def add_first(value)
-      raise NotImplementedError
-    end
+  # method to add a new node with the specific data value in the linked list
+  # insert the new node at the beginning of the linked list
+  # Time Complexity: O(1)
+  # Space Complexity O(1)
+  def add_first(value)
+    @head = Node.new(value, @head) # i don't fully understand why next_node is @head

```

Comment:
You're setting next_node to the **old** value of `@head`.  So the new node links to the start of the old list.